datacatalog/quickstart/src/main/java/com/example/datacatalog/CreateFilesetEntry.java

```diff
@@ -0,0 +1,171 @@
+    createEntry(projectId, entryGroupId, entryId);
+  }
+
+  // Create Fileset Entry.
+  public static void createEntry(String projectId, String entryGroupId, String entryId) {
+    // Currently, Data Catalog stores metadata in the us-central1 region.
+    String location = "us-central1";
+
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (DataCatalogClient dataCatalogClient = DataCatalogClient.create()) {
+
+      // 1. Environment cleanup: delete pre-existing data.
+      // Delete any pre-existing Entry with the same name
+      // that will be used in step 3.
+      try {
+        dataCatalogClient.deleteEntry(
+            EntryName.of(projectId, location, entryGroupId, entryId).toString());
+      } catch (PermissionDeniedException | NotFoundException e) {
+        // PermissionDeniedException or NotFoundException are thrown if
+        // Entry does not exist.
+        System.out.println("Entry does not exist.");
+      }
+
+      // Delete any pre-existing Entry Group with the same name
+      // that will be used in step 2.
+      try {
+        dataCatalogClient.deleteEntryGroup(
+            EntryGroupName.of(projectId, location, entryGroupId).toString());
+      } catch (PermissionDeniedException | NotFoundException e) {
+        // PermissionDeniedException or NotFoundException are thrown if
+        // Entry Group does not exist.
+        System.out.println("Entry Group does not exist.");
+      }
+
+      // 2. Create an Entry Group.
+      // Construct the EntryGroup for the EntryGroup request.
+      EntryGroup entryGroup =
+          EntryGroup.newBuilder()
+              .setDisplayName("My Fileset Entry Group")
+              .setDescription("This Entry Group consists of ....")
+              .build();
+
+      // Construct the EntryGroup request to be sent by the client.
+      CreateEntryGroupRequest entryGroupRequest =
+          CreateEntryGroupRequest.newBuilder()
+              .setParent(LocationName.of(projectId, location).toString())
+              .setEntryGroupId(entryGroupId)
+              .setEntryGroup(entryGroup)
+              .build();
+
+      // Use the client to send the API request.
+      EntryGroup entryGroupResponse = dataCatalogClient.createEntryGroup(entryGroupRequest);
+
+      System.out.printf("\nEntry Group created with name: %s\n", entryGroupResponse.getName());
+
+      // 3. Create a Fileset Entry.
+      // Construct the Entry for the Entry request.
+      Entry entry =
+          Entry.newBuilder()
+              .setDisplayName("My Fileset")
+              .setDescription("This fileset consists of ....")
+              .setGcsFilesetSpec(
+                  GcsFilesetSpec.newBuilder().addFilePatterns("gs://my_bucket/*").build())
```

Comment:
You might want to use a real bucket here, such as `gs://cloud-samples-data`

Reply:
according to the docs, underscore should be accepted, I filled a bug for this, but using a real bucket is better, made the change, thanks.